cli/det_sim.py

```diff
@@ -80,43 +127,83 @@ def recv_and_process():
 
     ts_offset = 0
 
-    @asyncio.coroutine
-    def sim_data():
+    _, udp = await loop.create_datagram_endpoint(
+        ListenAndSend, local_addr=('localhost', 0x7D00))
+
+    # this is not actually used (yet)
+    await loop.create_datagram_endpoint(
+        RegisterPort, local_addr=('localhost', 0x7D01))
+
+    async def sim_data():
         nonlocal ts_offset
         state[FRAMENUMREG] += 1
         num_per_msg = np.random.poisson(N, size=n_msgs)
         ts_offset = 0
+        udp_packet_count = 0
+        tail = []
         for num in num_per_msg:
             payload, ts_offset = make_sim_payload(num,
                                                   n_chips, n_chans,
                                                   tick_gap,
                                                   ts_offset)
-            yield from publisher.send_multipart([b'data', payload])
-
-        yield from publisher.send_multipart([b'meta',
-                                             np.array([state[FRAMENUMREG], 0],
-                                                      dtype=np.uint32)])
+            await publisher.send_multipart([b'data', payload])
+            if udp_packet_count == 0:
+                # special case packet 0
+                tail = payload
+                head, tail = tail[:510], tail[510:]
+                header = struct.pack('!IIIxxxx',
+                                     udp_packet_count,
+                                     0xfeedface,
+                                     state[FRAMENUMREG])
+                udp.send(b''.join((header,
+                                   bytes(head))))
+                udp_packet_count += 1
+            else:
+                first_head = 511 - len(tail)
+                head = (tail, payload[:first_head])
+                tail = payload[first_head:]
+                header = struct.pack('!Ixxxx', udp_packet_count)
+                udp.send(b''.join((header,
+                                   bytes(head[0]),
```

Comment:
these are all `np.ndarrays` for me and had to change to `head[0].tobytes()` etc.
do you know why this worked for you? I have numpy 1.14

Reply:
What version of python?  There was a subtle bug around this sort of thing that we hit in caproto (@danielballan and @klauer may remember the exact details better than I do).

Reply:
I'm using 3.6.0. Not sure what it is. Is it safer to call `.tobytes` anyway?

Reply:
Reference: https://github.com/NSLS-II/caproto/issues/48

Reply:
update to 3.6.2 and that should be fixed.